packages/gestalt/src/ModalAlert/Action.js

```diff
@@ -1,66 +1,60 @@
 // @flow strict
 import { type Node } from 'react';
 import Button from '../Button.js';
+import ButtonLink from '../ButtonLink.js';
 
-export type ActionDataType = {|
+type LinkDataType = {|
   accessibilityLabel: string,
+  dataTestId?: string,
   disabled?: boolean,
-  href?: string,
+  href: string,
   label: string,
-  onClick?: ({|
-    event:
-      | SyntheticMouseEvent<HTMLButtonElement>
-      | SyntheticMouseEvent<HTMLAnchorElement>
-      | SyntheticKeyboardEvent<HTMLAnchorElement>
-      | SyntheticKeyboardEvent<HTMLButtonElement>,
-    dangerouslyDisableOnNavigation: () => void,
-  |}) => void,
+  onClick?: $ElementType<React$ElementConfig<typeof ButtonLink>, 'onClick'>,
   rel?: 'none' | 'nofollow',
+  role: 'link',
   target?: null | 'self' | 'blank',
+  type: string,
+|};
+type ButtonDataType = {|
+  accessibilityLabel: string,
+  dataTestId?: string,
+  disabled?: boolean,
+  label: string,
+  onClick?: $ElementType<React$ElementConfig<typeof Button>, 'onClick'>,
+  role?: 'button',
+  type: string,
 |};
 
-type Props = {| ...ActionDataType, dataTestId?: string, type: string |};
+type Props = LinkDataType | ButtonDataType;
 
-export default function ModalAlertAction({
-  accessibilityLabel,
-  dataTestId,
-  disabled,
-  label,
-  onClick,
-  href,
-  rel,
-  target,
-  type,
-}: Props): Node {
+export default function ModalAlertAction({ dataTestId, type, ...props }: Props): Node {
```

Comment:
nit:
can we add in destructuring all the common / base ones: for example accessibility

this can be a cleanup followup

Reply:
DONE